http-testkit-specs2/src/main/scala/com/wix/e2e/http/matchers/internal/HeaderMatching.scala

```diff
@@ -31,23 +68,28 @@ trait HeaderMatching[T <: HttpMessage] {
   protected def specialHeaders: Map[String, String] = Map.empty
 
   def haveAnyHeadersOf(headers: (String, String)*): Matcher[T] =
-    haveHeaderInternal(headers, _.identical.nonEmpty,
-      res => s"Could not find header [${res.missing.map(_._1).mkString(", ")}] but found those: [${res.extra.map(_._1).mkString(", ")}]")
+    haveHeaderInternal(headers,
+      { case res if res.identical.isEmpty => s"Could not find header [${res.missing.names}] but found those: [${res.extra.names}]" })
 
   def haveAllHeadersOf(headers: (String, String)*): Matcher[T] =
-    haveHeaderInternal(headers, _.missing.isEmpty,
-      res => s"Could not find header [${res.missing.map(_._1).mkString(", ")}] but found those: [${res.identical.map(_._1).mkString(", ")}].")
+    haveHeaderInternal(headers,
+      { case res if res.missing.nonEmpty || res.contentDiff.nonEmpty => s"Could not find header [${res.missing.names}] but found those: [${res.identical.names}]." }
```

Comment:
@noam-almog I actually think this is confusing to only show the identical names.
there could be a typo and the header name mismatch for some reason. in a way i think it might be more beneficial to explicitly say what was the same, what was missing, what was extra.
it could even be wiser to make the same error message for all those 3 matchers, since it would always give the same message - no surprises.

Reply:
can you show an example of what is confusing ? do you want to remove the identical names section of the message ?

Reply:
for example, I expect (ABC -> "hi") instead the response came back with (ACB -> "hi"). it will show:
`Could not find header [ABC] but found those: [].`
which really does not really help me identify the issue, don't you think?
As I said, I think it would be best to always give the entire information:
`Found matching headers [...]. Found Matching headers with content difference [...]. missing headers are [..]. extra headers are [...].`
wording is up to you. also, if it's too verbose we can omit sections that are empty.

Reply:
first, the first message displays an empty list which is a bug and needs to be fix anyhow.
as for your comments, looks better, pay attention when you present those errors show them with the order someone will be expect to see them. if you throw just large amount of text to the dev it will be too much.
i would go with:

```
Found Matching headers with content difference [...]

missing headers are [..]

extra headers are [...]

Matching headers [...]. . .  

```